sqlmesh/dbt/basemodel.py

```diff
@@ -217,12 +217,33 @@ def tests_ref_source_dependencies(self) -> Dependencies:
         dependencies.macros = []
         return dependencies
 
+    def check_for_circular_test_refs(self, context: DbtContext) -> None:
+        for test in self.tests:
+            for ref in test.dependencies.refs:
+                model = context.refs[ref]
+                if ref == self.name or ref in self.dependencies.refs:
+                    continue
+                elif self.name in model.dependencies.refs:
```

Comment:
Does this mean that it only checks circular references 1 level deep? What about a graph like this:
```
A <- B <- C
```
and then model A has a relationship test with model C?